bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/TabViewPagerActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.yelp.android.bentosampleapp
+
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.viewpager.widget.ViewPager
+import com.google.android.material.tabs.TabLayout
+import com.yelp.android.bento.componentcontrollers.TabViewPagerComponentController
+import com.yelp.android.bento.core.Component
+import com.yelp.android.bentosampleapp.components.LabeledComponent
+import java.util.ArrayList
+
+class TabViewPagerActivity: AppCompatActivity(),
+        ViewPager.OnPageChangeListener{
+
+    private lateinit var componentController: TabViewPagerComponentController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        componentController = TabViewPagerComponentController()
+        setContentView(R.layout.activity_tab_viewpager)
+
+        val mTabLayout: TabLayout = findViewById(R.id.tab_layout)
+
+        val mViewPager: ViewPager = findViewById(R.id.view_pager)
+        mViewPager.addOnPageChangeListener(this)
+
+        mTabLayout.setupWithViewPager(mViewPager)
+
+        componentController.setViewPager(mViewPager)
+        addTabViewPagerComponent()
+    }
+
+    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+        Toast.makeText(this, "Tab $position selected", Toast.LENGTH_LONG).show()
+    }
+
+    override fun onPageScrollStateChanged(state: Int) {}
+
+    override fun onPageSelected(position: Int) {}
+
+    private fun addTabViewPagerComponent() {
```

Comment:
This function is not very kotliny. Instead of using `ArrayList` we should use `listOf` or `mutableListOf`. We can also use list operations to make this all more concise. 
```kotlin
val titles = listOf("Tab 1", "Tab 2")
componentController.titles = titles
componentController.addAll(titles.map { LabeledComponent("$it content") })
```